learn/appendix_opengl/KHR_debug.md

```diff
@@ -116,6 +116,23 @@ void GL.ObjectPtrLabel(
 );
 ```
 
+> [!WARNING]
+> The `length` parameter for most of the functions given are intended for C
+> users and is in ASCII characters, not UTF-16 characters. This causes a small
+> disparity between the length of a string in C#, and it's length in bytes after
+> marshalling between UTF-16 and ASCII/UTF-8 characters.
+>
+> Although in some cases (particularly where only 7-bit ASCII characters are
+> used) the number of characters in the C# string and the number of characters
+> in the C string are going to match, the most appropriate way would be to
+> measure the length of the string in bytes using
+> `System.Text.Encoding.UTF8.GetByteCount()`. However, that still takes
```

Comment:
I don't think we need to mention `UTF8.GetByteCount()` here, it will only confuse people and create bad code. The marshalling is going to null terminate anyways so `-1` is the "correct"/"best" way to handle this.

Reply:
Well, you complained that I said -1 would be the best method last time, so I added that as extra info.

Reply:
Damn, I had to go back and read that. Totally forgot I said that. 
I think I might have just 180ed on this...

I think we should say that `string.length` is not always the correct thing to pass here, and that `-1` is the best alternative. While not mentioning `GetByteCount()` (people who care know enough to figure that out themselves).